pkg/oc/cli/cmd/newapp.go

```diff
@@ -385,13 +390,42 @@ func (o *NewAppOptions) RunNewApp() error {
 			fmt.Fprintf(out, "%sTrack installation of %s with '%s logs %s'.\n", indent, installing[i].Name, o.BaseName, installing[i].Name)
 		}
 	case len(result.List.Items) > 0:
+		//if we don't find a route we give a message to expose it
+		containRoute := false
+		for _, i := range result.List.Items {
+			switch i.(type) {
+			case *routeapi.Route:
+				containRoute = true
+			}
+		}
+		if !containRoute {
+			//we if don't have any routes, but we have services - we suggest commands to expose those
+			svc := getServices(result.List.Items)
+			if len(svc) > 0 {
+				fmt.Fprintf(out, "%sApplication is not exposed. You can expose services to the outside world by executing one or more of the commands below:\n", indent)
+				for _, s := range svc {
+					fmt.Fprintf(out, "%s '%s %s svc/%s' \n", indent, o.BaseName, ExposeRecommendedName, s.Name)
+				}
+			}
+		}
 		fmt.Fprintf(out, "%sRun '%s %s' to view your app.\n", indent, o.BaseName, StatusRecommendedName)
 	}
 	return nil
 }
 
 type LogsForObjectFunc func(object, options runtime.Object, timeout time.Duration) (*restclient.Request, error)
 
+func getServices(items []runtime.Object) []*kapi.Service {
+	var svc []*kapi.Service
+	for _, i := range items {
+		switch i.(type) {
```

Comment:
no need to change it, but for reference, you can also do `if s, ok := i.(*kapi.Service); ok { svc = append(svc, s) }`